module/x/gravity/keeper/ethereum_event_handler.go

```diff
@@ -121,3 +78,98 @@ func (a EthereumEventProcessor) Handle(ctx sdk.Context, eve types.EthereumEvent)
 	}
 	return nil
 }
+
+func (a EthereumEventProcessor) verifyERC20DeployedEvent(ctx sdk.Context, event *types.ERC20DeployedEvent) error {
+	if existingERC20, exists := a.keeper.getCosmosOriginatedERC20(ctx, event.CosmosDenom); exists {
+		return sdkerrors.Wrapf(
+			types.ErrInvalid,
+			"ERC20 token %s already exists for denom %s", existingERC20.Hex(), event.CosmosDenom,
+		)
+	}
+
+	// We expect that all Cosmos-based tokens have metadata defined. In the case
+	// a token does not have metadata defined, e.g. an IBC token, we successfully
+	// handle the token under the following conditions:
+	//
+	// 1. The ERC20 name is not equal to the token's denomination. Otherwise, this
```

Comment:
This is incorrect. We check that the ERC20 name IS equal to the denom, since this is the only identifier we have without metadata.

Reply:
Updated. Good catch.